app/router.js

```diff
@@ -12,28 +12,56 @@ import {
   removeLabel,
   reopenIssue,
   requestReviewers,
+  reportError,
   transferIssue,
 } from "./github.js";
 import { getAuthToken } from "./auth.js";
 import { extractCommaSeparated, extractUsersAndTeams } from "./converters.js";
+import { labelEnabled, transferEnabled } from "./command-enabled.js";
+import { defaultConfig } from "./default-config.js";
+
+const { Octokit } = require("@octokit/core");
+const { config, composeConfigGet } = require("@probot/octokit-plugin-config");
 
 export async function router(auth, id, payload, verbose) {
   const sourceRepo = payload.repository.name;
   const transferMatches = transferMatcher(payload.comment.body);
   const actorRequest = `as requested by ${payload.sender.login}`;
+
+  const authToken = await getAuthToken(auth, payload.installation.id);
+  const octokit = new Octokit({ auth: authToken });
+
+  // TODO validate against schema
+  const { config } = await octokit.config.get({
+    owner: payload.repository.owner.login,
+    repo: sourceRepo,
+    path: ".github/comment-ops.yml",
+    defaults: defaultConfig,
+  });
+
   if (transferMatches) {
-    const targetRepo = transferMatches[1];
-    console.log(
-      `${id} Transferring issue ${payload.issue.html_url} to repo ${targetRepo} ${actorRequest}`
-    );
-    await transferIssue(
-      await getAuthToken(auth, payload.installation.id),
-      payload.repository.owner.login,
-      sourceRepo,
-      targetRepo,
-      payload.issue.node_id
-    );
-    return;
+    const enabled = await transferEnabled(octokit, config);
+
+    if (enabled) {
+      const targetRepo = transferMatches[1];
+      console.log(
+        `${id} Transferring issue ${payload.issue.html_url} to repo ${targetRepo} ${actorRequest}`
+      );
+      await transferIssue(
+        await getAuthToken(auth, payload.installation.id),
```

Comment:
Your generating multiple auth tokens.

The token should be valid up to 10 minutes.

One option is to actually cache and rotate appropriately

Reply:
aware :) will refactor before merge, either need to call each config load inside the if block or do all matches separately at the beginning and then check the results.

I'm trying to avoid loading config if the comment doesn't match in the first place.
I'll make sure no double auth once I'm more sure on the design 😄

Reply:
You could move all the matchers to the top and keep them in an array.

One you can loop through all matches. Two you can skip if match count is zero.

Reply:
Actually you could consider a more generic approach.
That would allow for easy expansion of new matches and call them appropriately into a generic method when you loop through them, where they call their special function that is custom to each matcher.

Reply:
will think about it :)

Reply:
I've made a start on it, which has allowed me to fix the extra auth token calling

Reply:
I used

```
const runCommands = Object.values(commands).filter((command) => command.matches)
for (const command of runCommands) {
  command.run(needed properties....)
}
```

because I want to be able to report when commands are not enabled and I don't want to do multiple calls to enabled as it may make remote calls for permissions.

I'm undecided if enabled and permission is the same thing yet, but I'll probably merge the PR without introducing permissions and do that in a follow up.

This is looking a lot better now and I think just needs the enabled commands to be filled out for all